core/src/main/scala/kafka/consumer/BaseConsumer.scala

```diff
@@ -54,6 +57,8 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
   import org.apache.kafka.clients.consumer.KafkaConsumer
 
   val consumer = new KafkaConsumer[Array[Byte], Array[Byte]](consumerProps)
+  val offsets = new HashMap[TopicPartition, Long]()
```

Comment:
How about simply using a set of TopicPartition to cache all partitions? If invoking `records` and `partitions` I also have to employ multiple iterators to track the current partition and the record streams for that partition. Which one do you prefer?